DailyQuestionHistory/2022-08-24.md

```diff
@@ -0,0 +1,75 @@
+### [17. Letter Combinations of a Phone Number](https://leetcode.com/problems/letter-combinations-of-a-phone-number/)
+Kang
+```
+/**
+ * @param {string} digits
+ * @return {string[]}
+ */
+var letterCombinations = function(digits) {
+    if(digits.length === 0){
+        return [];
+    }
+    var res = [];
+    var map = {
+        '2': ["a","b","c"],
+        '3': ["d","e","f"],
+        '4': ["g","h","i"],
+        '5': ["j","k","l"],
+        '6': ["m","n","o"],
+        '7': ["p","q","r","s"],
+        '8': ["t","u","v"],
+        '9': ["w","x","y","z"]
+    };
+    var len = digits.length;
+    for(var letters of map[digits.charAt(0)]){
+        for(var letter of letters){
```

Comment:
letters is already a single alphabet. Why do we need to double loop it?

Reply:
`for(var letter of map[digits.charAt(0)]){
            build(letter, 0);
        }
`
is enough